sonic_installer/main.py

```diff
@@ -530,10 +532,17 @@ def install(url, force, skip_migration=False, skip_package_migration=False,
             echo_and_log('Error: Failed to set image as default', LOG_ERR)
             raise click.Abort()
     else:
+        # Verify that the binary image is secureboot
+        if not bootloader.verify_secureboot_image(image_path) and not force_secureboot:
+            echo_and_log("Image file '{}' is not a secureboot image.\n".format(url) +
+                "If you are sure you want to install this image, use --force_secureboot.\n" +
+                "Aborting...", LOG_ERR)
+            raise click.Abort()
+
         # Verify that the binary image is of the same type as the running image
```

Comment:
>Verify that the binary image is of the same type as the running image [](http://example.com/codeflow?start=10&length=69)

The comment is not accurate. Has nothing to do with "the running image".

Reply:
what about 
Verify the binary image is of the same platform type as the running platform.